internal/driver/driver.go

```diff
@@ -140,8 +170,16 @@ func (d *Driver) Start() error {
 	// Make sure the paths of existing devices are up-to-date.
 	go d.RefreshAllDevicePaths()
 
+	if !d.enableRtspServer {
+		d.lc.Info("Rtsp server is disabled")
+		return nil
+	}
+
 	d.wg.Add(1)
-	go d.StartRTSPCredentialServer()
+	go func() {
+		defer d.wg.Done()
+		d.StartRTSPCredentialServer()
```

Comment:
I don't see that you removed the `d.wg.Done()` from `StartRTSPCredentialServer`? If not, please remove it.